ipamd/datastore/data_store.go

```diff
@@ -41,19 +41,23 @@ var ErrUnknownPodIP = errors.New("datastore: pod using unknown ip address")
 
 // ENIIPPool contains ENI/IP Pool information
 type ENIIPPool struct {
-	createTime            time.Time
-	lastUnassignedTime    time.Time
-	isPrimary             bool
-	id                    string
-	deviceNumber          int
-	assignedIPv4Addresses int
-	ipv4Addresses         map[string]*AddressInfo
+	createTime         time.Time
+	lastUnAssignedTime time.Time
+	// IsPrimary indicates whether eni is a primary eni
+	IsPrimary bool
+	id        string
+	// DeviceNumber is the device number of eni
+	DeviceNumber int
+	// AssignedIPv4Addresses is the number of ip addesses already been assigned
+	AssignedIPv4Addresses int
+	// IPv4Addresses shows whether each address is assigned
+	IPv4Addresses map[string]*AddressInfo
 }
 
 // AddressInfo contains inforation about an IP
 type AddressInfo struct {
 	address        string
```

Comment:
Please document the form of this string.  Is it redundant with the key of the map to these structs?

Reply:
The form of this string is the dot notation of IP address.  Making address as part of AddressInfo struct can make AddressInfo re-used by other part of code,

Reply:
The form should be documented in the code; suggest: "IP address strings must be in dot-decimal notation with no leading zeroes and no whitespace (eg: '10.1.0.253')"

Note: there was an attempt to [standardize text representation](https://tools.ietf.org/html/draft-main-ipaddr-text-rep-00) via IETF RFC, though it was never formally accepted.  As you can see from that doc, there is historical ambiguity in how to represent IP addresses as text.

Reply:
ok